e2e/utils/pages/homepage.page.ts

```diff
@@ -0,0 +1,64 @@
+import {
+	HERO_ANIMATION_CANVAS,
+	LOGIN_BUTTON,
+	TOKENS_SKELETONS_INITIALIZED
+} from '$lib/constants/test-ids.constant';
+import { type InternetIdentityPage } from '@dfinity/internet-identity-playwright';
+import { type Page } from '@playwright/test';
+import { HOMEPAGE_URL, LOCAL_REPLICA_URL } from '../constants/e2e.constants';
+
+export class Homepage {
+	protected readonly _page: Page;
+
+	constructor(page: Page) {
+		this._page = page;
+	}
+
+	protected async hideHeroAnimation(): Promise<void> {
+		await this._page
+			.getByTestId(HERO_ANIMATION_CANVAS)
+			.evaluate((element) => (element.style.display = 'none'));
+	}
+
+	get page(): Page {
+		return this._page;
+	}
+
+	async goto(): Promise<void> {
+		await this._page.goto(HOMEPAGE_URL);
+	}
+
+	async waitForLoggedOut(): Promise<void> {
```

Comment:
I'm not sure so just asking, shouldn't this function finds place in a child not in the parent class?

Reply:
I consider `Homepage` class as a `HomepageLoggedOut`. That's why I kept `waitForLoggedOut` there and have `waitForLoggedIn` in `HomepageLoggedIn`. Now that you asked it, I think it makes sense to rename in the code to keep the intention clear. Let me know what you think.

Reply:
It's a bit confusing to me but, I'm also maybe just picky 😅. Why not having `Homepage` as an `abstract class` and then two implementation `HomepageLoggedOut` and `HomepageLoggedIn`?

Reply:
Yep, was confusing indeed - I now update it the way you suggested.